src/pathCalculator/PathCalculatorClient.java

```diff
@@ -0,0 +1,107 @@
+package pathCalculator;
+
+import sample.Map.MapController;
+import sample.Map.MapModel;
+import sample.StaticClasses.Point;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PathCalculatorClient {
+
+    private static PathCalculatorClient instance;
+
+    private static InetAddress DEFAULT_SERVER_HOST = InetAddress.getLoopbackAddress();
+    private static int DEFAULT_PORT = 15000;
+    public static final String SEPERATOR = "~";
+
+
+    private boolean isConnected;
+    private InputStream inputStream;
+    private OutputStream outputStream;
+
+    // Do not allow external instances of Path Calculator
+    private PathCalculatorClient() {
+        isConnected = false;
+        inputStream = null;
+        outputStream = null;
+    }
+
+    public static PathCalculatorClient getInstance() {
+        if (instance == null)
+            instance = new PathCalculatorClient();
+        return instance;
+    }
+
+    public boolean isConnected() {
+        return isConnected;
+    }
+
+    public void connect() throws IOException {
+        Socket socket = new Socket(DEFAULT_SERVER_HOST, DEFAULT_PORT);
+        inputStream = socket.getInputStream();
+        outputStream = socket.getOutputStream();
+        isConnected = true;
+    }
+
+    public List<Point> requestSolution(MapModel model) throws IOException {
+        byte[] request = buildRequestByteArray(model);
+        this.outputStream.write(request);
+        byte[] buffer = new byte[1024000];
+        int read = inputStream.read(buffer);
+        String msg = new String(buffer, 0, read);
+        return responseAnswers(msg);
+    }
+
+    private byte[] buildRequestByteArray(MapModel model) {
+        Point startPoint = model.toMapPoint(model.getStartPosition()),
```

Comment:
toMapPoint is a private method. 
Change it to public?

Reply:
yes